packages/react/src/date-picker/DatePicker.tsx

```diff
@@ -197,22 +213,29 @@ export const DatePicker = ({
 				onBlur={onInputBlur}
 				onChange={onInputChange}
 				buttonRef={triggerRef}
-				buttonOnClick={toggleCalendar}
+				buttonOnClick={() => {
+					toggleCalendar();
+					setHasCalendarOpened(true);
+				}}
 				buttonAriaLabel={getDateInputButtonAriaLabel(inputValue)}
 			/>
 			<CalendarProvider yearRange={yearRange}>
-				<Popover {...popover.getPopoverProps(isCalendarOpen)}>
-					{isCalendarOpen && (
-						<CalendarSingle
-							initialFocus
-							selected={valueAsDateOrUndefined}
-							onSelect={onSelect}
-							defaultMonth={defaultMonth}
-							numberOfMonths={1}
-							disabled={disabledCalendarDays}
-						/>
-					)}
-				</Popover>
+				{/* We duplicate the Popover + Calendar as a workaround for a bug that scrolls the page to the top on initial open of the calandar - https://github.com/gpbl/react-day-picker/discussions/2059 */}
+				{hasCalendarOpened ? ( // If the calendar has opened at least once, we conditionally render the Popover + children to prevent the scroll-to-top bug
+					isCalendarOpen && (
+						<Popover {...popoverProps}>
+							<CalendarSingle {...calendarProps} />
+						</Popover>
+					)
+				) : (
+					// If the calendar has _not_ opened at least once, we conditionally render only the children of the Popover, i.e. the Calendar to prevent the UI jumping about everytime the calendar is opened
+					<Popover
+						{...popoverProps}
+						css={{ visibility: isCalendarOpen ? 'visible' : 'hidden' }}
```

Comment:
Should be able to do this: `visibility={isCalendarOpen}`

Reply:
🤯

Reply:
I had to do this but it's a bit cleaner 🙂

```ts
visibility={isCalendarOpen ? 'visible' : 'hidden'}
```